### PR TITLE
Remove `divisible`, copy documentation to `is_divisible_by` instead

### PR DESCRIPTION
**Pull request by lgoettgens**
Resolves https://github.com/oscar-system/Oscar.jl/issues/3178 by removing `divisible` and promoting `is_divisible_by` with a fixed docstring instead. (as suggested by @fieker in https://github.com/oscar-system/Oscar.jl/issues/3178#issuecomment-1887310107)

This function is already used throughout AbstractAlgebra. Thus I removed the duplicate implementation for `(::Integer, ::Integer)` here in Nemo.

cc @JohnAAbbott

### CODE REVIEW

**Review by fieker — APPROVED** [review]
Shall we also remove divides? Or is the meaning there clear?

**Comment by lgoettgens** [comment]
I think divides additionally returns the quotient.
I'll put it into the renaming hackmd for further discussion. But that's for the future

**Comment by fieker** [comment]
On Thu, Jan 11, 2024 at 08:10:57AM -0800, Lars Göttgens wrote:
> I think divides additionally returns the quotient.
> I'll put it into the renaming hackmd for further discussion. But that's for the future
> 
Thanks

**Comment by thofma** [comment]
`divides` is `is_divisible_by` with additional quotient. Not sure if this makes sense, we decided upon the names here: https://github.com/Nemocas/Nemo.jl/issues/862#issuecomment-669289362. It would be more logical to do `is_divisible_by_with_quotient`, but decided against it in https://github.com/Nemocas/Nemo.jl/issues/862#issuecomment-669306021.

**Comment by lgoettgens** [comment]
> `divides` is `is_divisible_by` with additional quotient. Not sure if this makes sense, we decided upon the names here: [#862 (comment)](https://github.com/Nemocas/Nemo.jl/issues/862#issuecomment-669289362). It would be more logical to do `is_divisible_by_with_quotient`, but decided against it in [#862 (comment)](https://github.com/Nemocas/Nemo.jl/issues/862#issuecomment-669306021).

Thanks for the ref. I added it to the hackmd.

This PR is good to go then, right?

**Comment by lgoettgens** [comment]
Ah, CI is failing, but it seems to be some kind of GAP thing:
```julia
OscarInterface: Error During Test at /home/runner/.julia/packages/Oscar/1Moid/test/GAP/oscarinterface.jl:5
  Test threw exception
  Expression: GAP.Globals.TestDirectory(GAP.Globals.DirectoriesPackageLibrary(GAP.Obj("OscarInterface"), GAP.Obj("tst")))
  Error thrown by GAP: Error, UndefVarError: `divisible` not defined
```

summoning our GAP expert @fingolfin since I don't see this function used in the GAP tests at all...

**Comment by lgoettgens** [comment]
@fingolfin wanted to look at this in the afternoon

**Comment by fingolfin** [comment]
Spent the afternoon in renaming discussions... But now I had a look. And it is exactly as I suspected, the error comes from a call into Hecke. Here is condensed version of some of the code being run,
```
julia> K = GAP.Globals.K
Number field with defining polynomial x^4 + 4*x^3 - 100*x^2 - 208*x + 1744
  over rational field

julia> maximal_order(K)
ERROR: UndefVarError: `divisible` not defined
Stacktrace:
 [1] pmaximal_overorder_at(O::NfOrd, primes::Vector{ZZRingElem})
   @ Hecke ~/.julia/packages/Hecke/HCQ1K/src/NumFieldOrd/NfOrd/MaxOrd/MaxOrd.jl:113
 [2] new_maximal_order(O::NfOrd; index_divisors::Vector{ZZRingElem}, disc::ZZRingElem, ramified_primes::Vector{ZZRingElem})
   @ Hecke ~/.julia/packages/Hecke/HCQ1K/src/NumFieldOrd/NfOrd/MaxOrd/MaxOrd.jl:200
 [3] (::Hecke.var"#2069#2070"{Vector{ZZRingElem}, AnticNumberField})()
   @ Hecke ~/.julia/packages/Hecke/HCQ1K/src/NumFieldOrd/NfOrd/MaxOrd/MaxOrd.jl:44
 [4] get!(default::Hecke.var"#2069#2070"{Vector{ZZRingElem}, AnticNumberField}, h::Dict{Symbol, Any}, key::Symbol)
   @ Base ./dict.jl:479
 [5] get_attribute!(f::Function, G::AnticNumberField, attr::Symbol)
   @ AbstractAlgebra ~/.julia/packages/AbstractAlgebra/R29qD/src/Attributes.jl:230
 [6] MaximalOrder(K::AnticNumberField; discriminant::ZZRingElem, ramified_primes::Vector{ZZRingElem})
   @ Hecke ~/.julia/packages/Hecke/HCQ1K/src/NumFieldOrd/NfOrd/MaxOrd/MaxOrd.jl:42
 [7] MaximalOrder(K::AnticNumberField)
   @ Hecke ~/.julia/packages/Hecke/HCQ1K/src/NumFieldOrd/NfOrd/MaxOrd/MaxOrd.jl:41
 [8] top-level scope
   @ REPL[5]:1
```

**Comment by lgoettgens** [comment]
Ah okay, so Hecke doesn't like the deprecation somehow. Let me just mark this breaking then.

File: src/Deprecations.jl [hunk 1]
```diff
@@ -115,3 +115,8 @@
 @deprecate roots(f::QQPolyRingElem, R::T) where {T<:Union{fqPolyRepField,fpField}} roots(R, f)
 
 @deprecate factor(f::QQPolyRingElem, R::T) where {T<:Union{fqPolyRepField,fpField}} factor(R, f)
+
+# Deprecated in 0.39.*
+
+@deprecate divisible(x::ZZRingElem, y::Int) is_divisible_by(x, y)
```
Comment (lgoettgens) [inline src/Deprecations.jl:121]:
```suggestion
@deprecate divisible(x::Int, y::Int) is_divisible_by(x, y)
@deprecate divisible(x::ZZRingElem, y::Int) is_divisible_by(x, y)
```

**Pull request merged**